src/commonTest/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandlerTestsCommon.kt

```diff
@@ -925,37 +933,91 @@ class IncomingPaymentHandlerTestsCommon : LightningTestSuite() {
     }
 
     @Test
-    fun `mpp success then additional HTLC`() = runSuspendTest {
+    fun `mpp success then HTLC local replay`() = runSuspendTest {
+        val (amount1, amount2) = listOf(12_000.msat, 50_000.msat)
+        val (channelId1, channelId2) = listOf(randomBytes32(), randomBytes32())
+        val totalAmount = amount1 + amount2
+        val (paymentHandler, incomingPayment, paymentSecret) = createFixture(totalAmount)
+        val (htlc1, htlc2) = listOf(
+            makeUpdateAddHtlc(876, channelId1, paymentHandler, incomingPayment.paymentHash, makeMppPayload(amount1, totalAmount, paymentSecret)),
+            makeUpdateAddHtlc(2, channelId2, paymentHandler, incomingPayment.paymentHash, makeMppPayload(amount2, totalAmount, paymentSecret)),
+        )
+
+        // Step 1 of 2:
+        // - Alice receives complete mpp set
+        run {
+            val result1 = paymentHandler.process(htlc1, TestConstants.defaultBlockHeight)
+            assertTrue { result1 is IncomingPaymentHandler.ProcessAddResult.Pending }
+            assertTrue { result1.actions.isEmpty() }
+
+            val result2 = paymentHandler.process(htlc2, TestConstants.defaultBlockHeight)
+            assertTrue { result2 is IncomingPaymentHandler.ProcessAddResult.Accepted }
+
+            val expected = setOf(
+                WrappedChannelEvent(channelId1, ChannelEvent.ExecuteCommand(CMD_FULFILL_HTLC(htlc1.id, incomingPayment.preimage, commit = true))),
+                WrappedChannelEvent(channelId2, ChannelEvent.ExecuteCommand(CMD_FULFILL_HTLC(htlc2.id, incomingPayment.preimage, commit = true))),
+            )
+            assertEquals(expected, result2.actions.toSet())
+        }
+
+        // Step 2 of 2:
+        // - Alice receives local replay of htlc1 for the invoice she already completed. Must be fulfilled.
+        run {
+            val result = paymentHandler.process(htlc1, TestConstants.defaultBlockHeight)
+            assertTrue { result is IncomingPaymentHandler.ProcessAddResult.Accepted }
+            val expected = WrappedChannelEvent(channelId1, ChannelEvent.ExecuteCommand(CMD_FULFILL_HTLC(htlc1.id, incomingPayment.preimage, commit = true)))
+            assertEquals(setOf(expected), result.actions.toSet())
+        }
+    }
+
+    @Test
+    fun `mpp success then new HTLC`() = runSuspendTest {
```

Comment:
nit: same for this one:
```suggestion
    fun `reject htlcs for already paid invoices`() = runSuspendTest {
```